rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/project/ProjectController.java

```diff
@@ -2169,17 +2170,31 @@ private Project convertToProject(Map<String, Object> requestBody) {
         mapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
         mapper.registerModule(sw360Module);
 
+        if (requestBody.containsKey(mapOfProjectFieldsToRequestBody.get(Project._Fields.VISBILITY))) {
+            try {
+                String visibility = (String) requestBody
+                        .get(mapOfProjectFieldsToRequestBody.get(Project._Fields.VISBILITY));
+                requestBody.put(mapOfProjectFieldsToRequestBody.get(Project._Fields.VISBILITY),
+                        visibility.toUpperCase());
+            } catch (IllegalArgumentException e) {
+                System.err.println("Error processing visibility field: " + e.getMessage());
+                System.err.println("Failed requestBody: " + requestBody);
+                throw e;
+            }
+        }
+
         if (requestBody.containsKey("linkedProjects")) {
             Map<String, Object> linkedProjects = (Map<String, Object>) requestBody.get("linkedProjects");
             linkedProjects.entrySet().stream().forEach(entry -> {
                 if (entry.getValue() instanceof String) {
-                    Map<String, Object> projectProjectRelationShip = new HashMap<String, Object>();
-                    projectProjectRelationShip.put("projectRelationship", entry.getValue());
-                    linkedProjects.put(entry.getKey(), projectProjectRelationShip);
+                    Map<String, Object> projectRelationshipMap = new HashMap<>();
+                    projectRelationshipMap.put("projectRelationship", entry.getValue());
+                    linkedProjects.put(entry.getKey(), projectRelationshipMap);
                 }
             });
-
         }
+
+        JsonNode jsonNode = mapper.valueToTree(requestBody);
```

Comment:
The variable `jsonNode` is not used anywhere. Is it required?

Reply:
not required. let me delete this.

Reply:
done.